src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
     vec::Vec,
 };
 
-/// Data structures used by a polynomial commitment scheme.
+// use rayon::prelude::*;
+
+/// A sponge-like random oracle for Fiat-Shamir transform usage.
+pub mod fiat_shamir_rng;
+
+/// Data structures for linear polynomial commitment schemes.
 pub mod data_structures;
 pub use data_structures::*;
 
 /// Errors pertaining to query sets.
 pub mod error;
 pub use error::*;
 
-use crate::rng::FiatShamirRng;
-
 /// A random number generator that bypasses some limitations of the Rust borrow
 /// checker.
 pub mod optional_rng;
 
-/// A polynomial commitment scheme based on the hardness of the
-/// discrete logarithm problem in prime-order groups.
-/// The construction is detailed in [[BCMS20]][pcdas].
-///
-/// [pcdas]: https://eprint.iacr.org/2020/499
+/// Domain extension of [`PolynomialCommitment`].
+pub mod domain_extended;
+pub use domain_extended::*;
+
+/// The [BCMS20](https://eprint.iacr.org/2020/499) variant of the dlog commitment scheme.
 pub mod ipa_pc;
 
-/// `QuerySet` is the set of queries that are to be made to a set of labeled polynomials/equations
-/// `p` that have previously been committed to. Each element of a `QuerySet` is a pair of
-/// `(label, (point_label, point))`, where `label` is the label of a polynomial in `p`,
-/// `point_label` is the label for the point (e.g., "beta"), and  and `point` is the field element
-/// that `p[label]` is to be queried at.
+/// `QuerySet` is the set of queries that are to be made to a set of labeled polynomials or linear combinations.
+///
+///  Each element of a `QuerySet` is a pair of `(label, (point_label, point))`, where
+///  * `label` is the label of a polynomial in `p`,
+///  * `point_label` is the label for the point (e.g., "beta"), and  and
+///  * `point` is the field element that `p[label]` is to be queried at.
```

Comment:
Thinking about it, the definition of a query set seems a bit strange to me. On the one hand, we refer to the polynomial by a label alone, but for the points we always provide the point itself (besides its label). We even do not check if the points are consistently chosen, i.e. that the same point is used for the same label. Maybe it is worth to overthink it.

Reply:
After a discussion with LucaG. and Nico, that also posed this problem, we agreed a possible solution would be:

`QueryMap<F> = BTreeMap<PointLabel, (PointValue, BTreeSet<PolyLabel>)>`

So, we associate to each point the polys in which it must be evaluated, and it's value (that will stay unique because we cannot have duplicate entries in the Map.

Currently, PointLabel and PolyLabel would be String.
Another minor improvement is to consider them as a generic type L: Debug + Ord + Eq + PartialEq
This will allow us, for instance, to use enums (instead of Strings) on Marlin for prover oracles and evaluation points (that we know a priori). This is just an aesthetic improvement

Reply:
Or, we have a separate map for labeling points?

Reply:
Also doable, but I don't really like passing around an additional data structure (we already have a huge list of arguments)

Reply:
Done